feat: use whiskers and add accent support

Anomalocaridid opened the pull request:
Generated the themes using whiskers and added variations for each accent.

Anomalocaridid committed:
feat: add template

Anomalocaridid committed:
chore: generate themes

Anomalocaridid committed:
docs: remove redundant FAQ question from README.md

uncenter commented on prismlauncher.tera:
```diff
@@ -0,0 +1,52 @@
+    - flavor
+    - accent
+  filename: "themes/{{flavor.identifier}}/{{accent}}/{{variant}}"
+tooltip_color: "dee5fc"
```
Why is this set separately? And it isn't a Catppuccin color?

Anomalocaridid replied:
I thought it was a Catppuccin color, but I checked https://catppuccin.com/palette and couldn't find it. Also, strangely, the theme currently uses the exact same color in all four flavors.

I figured setting it as a frontmatter variable was better than repeating it twice in the template. Also, this way its purpose is clearer and it is more obvious the template relies on an arbitrary hardcoded value since it is declared up front.

uncenter replied:
Where is it used? Like the color of tooltips? That should just be `text`. If it is the background of tooltips, it should be `crust` or `mantle`.

Anomalocaridid replied:
It appears to be used in the background color and text color of tooltips. I have replaced it with `crust` and `text` respectively.

uncenter commented on prismlauncher.tera:
```diff
@@ -0,0 +1,52 @@
+file_copyright_text: "2022 Catppuccin"
+license_identifier: "MIT"
+---
+{%- set palette = flavor.colors -%}
+{%- if variant == "theme.json" -%}
+{
+    "colors": {
+        "AlternateBase": "#{{palette.base.hex}}",
+        "Base": "#{{palette.mantle.hex}}",
+        "BrightText": "#{{palette.subtext1.hex}}",
+        "Button": "#{{palette.surface0.hex}}",
+        "ButtonText": "#{{palette.text.hex}}",
+        "Highlight": "#{{palette[accent].hex}}",
+        "HighlightedText": "#{{palette.base.hex}}",
+        "Link": "#{{palette.lavender.hex}}",
+        "Text": "#{{palette.text.hex}}",
+        "ToolTipBase": "#{{tooltip_color}}",
+        "ToolTipText": "#{{tooltip_color}}",
+        "Window": "#{{palette.base.hex}}",
+        "WindowText": "#{{palette.subtext1.hex}}",
+        "fadeAmount": 0.5,
+        "fadeColor": "#{{palette.overlay0.hex}}"
+    },
+    "name": "Catppuccin {{flavor.name}}",
+    "widgets": "Fusion"
+}
+{%- elif variant == "themeStyle.css" -%}
+/*
+ * SPDX-FileCopyrightText: {{file_copyright_text}}
+ *
+ * SPDX-License-Identifier: {{license_identifier}}
+ */
+
+QToolTip { 
+    color: #{{palette.text.hex}};
+    background-color: #{{palette.surface0.hex}};
+    border: 1px solid #{{palette.surface0.hex}}
+}
```
You can remove the `{% set palette ...` line and all the `palette.` parts as of v2.3.0`. GitHub sucks at suggestion diffs lol.
```suggestion
{%- if variant == "theme.json" -%}
{
    "colors": {
        "AlternateBase": "#{{base.hex}}",
        "Base": "#{{mantle.hex}}",
        "BrightText": "#{{subtext1.hex}}",
        "Button": "#{{surface0.hex}}",
        "ButtonText": "#{{text.hex}}",
        "Highlight": "#{{flavor.colors[accent].hex}}",
        "HighlightedText": "#{{base.hex}}",
        "Link": "#{{lavender.hex}}",
        "Text": "#{{text.hex}}",
        "ToolTipBase": "#{{tooltip_color}}",
        "ToolTipText": "#{{tooltip_color}}",
        "Window": "#{{base.hex}}",
        "WindowText": "#{{subtext1.hex}}",
        "fadeAmount": 0.5,
        "fadeColor": "#{{overlay0.hex}}"
    },
    "name": "Catppuccin {{flavor.name}}",
    "widgets": "Fusion"
}
{%- elif variant == "themeStyle.css" -%}
/*
 * SPDX-FileCopyrightText: {{file_copyright_text}}
 *
 * SPDX-License-Identifier: {{license_identifier}}
 */

QToolTip { 
    color: #{{text.hex}};
    background-color: #{{surface0.hex}};
    border: 1px solid #{{surface0.hex}}
}
```

uncenter commented on prismlauncher.tera:
```diff
@@ -0,0 +1,52 @@
+---
+whiskers:
+  version: 2.1.0
```
```suggestion
  version: 2.3.0
```

uncenter commented on themes/frappe/README.md:
We can/should delete these files. Additional previews should be added to the main README if anything, and seeing as these aren't generated by Whiskers there is no point in keeping them.

Anomalocaridid replied:
I have now removed the `preview.png`, `preview.png.license`, and `README.md` files in each flavor directory.

uncenter replied:
Ah well actually I think they are required files for each theme, according to the zips from the upstream releases. Not sure if we should keep them or not...

Anomalocaridid replied:
I guess that would depend on how much we care about continuing to upstream changes, which I honestly failed to consider when making this PR.

uncenter replied:
We will probably continue to update upstream, but by simply picking one flavor + accent combination to use there.

Anomalocaridid replied:
Makes sense. It looks like the current accent color is lavender, so we might as well stick with that so the preview images are still accurate.

In that case, I assume I should just move those files to the respective lavender theme folder for each flavor, right?

uncenter replied:
Uhh yeah I guess? Sounds fine to me.

Anomalocaridid replied:
Done.

uncenter commented on README.md:
```diff
@@ -53,7 +53,7 @@
 - A: Unfortunately some sections of Prism Launcher cannot be themed due to Qt using them.
 ---
 - Q: "Can I change the color?"
-- A: If you would like to change the accent/highlighted color, you can do this by modifying the `theme.json` file. Then editing the `"Highlight"` value to any color you prefer. You can also do this to other values if you would like.
+- A: If you would like to change any colors, you can do this by modifying the `theme.json` file. Then editing the the values to any color you prefer.
```
Not sure if any of this section should be here seeing as we already pre-generate all flavor + accent combinations? What other colors would need to be changed?

Anomalocaridid replied:
Good point. I have removed that question from the FAQ.

uncenter commented on prismlauncher.tera:
```diff
@@ -0,0 +1,50 @@
+SPDX-FileCopyrightText: {{file_copyright_text}}
+
+SPDX-License-Identifier: {{license_identifier}}
+{%- endif -%}
```
You should remove this extra `-` to make sure the files end with trailing newlines :)

Anomalocaridid replied:
Done

Anomalocaridid committed:
docs: move theme documentation to lavender for each flavor

Mqisty commented:
I no longer have the time to maintain this repo, thank you for the PR but I don't have any time to look over it. I am looking for someone else to take-over so your work doesn't go to waste.

sgoudham commented:
Little bit embarrassed that it's been over a year, but I'll look to review, test and merge this soon.

Anomalocaridid commented:
@sgoudham No worries, I understand you were probably busy with more pressing things.

Anomalocaridid committed:
ci: add whiskers check

Anomalocaridid commented:
I removed the justfile, updated the template to `^2.5.1`, added a CI whiskers check, squashed some of the commits to tidy the commit history a little, and rebased onto `main` just in case.

sgoudham reviewed (approved):
Going to merge this so that we have it under Whiskers, but I plan on doing some serious QoL updates to this repository. Primarily want to stop storing the generated output in source and instead upload zips to GitHub releases.

Sorry for the huge wait and thank you @Anomalocaridid!

Anomalocaridid commented:
No worries, it happens

sgoudham commented:
So I've been testing out the theme and making changes, and I honestly think this theme is better without accent support given how limited it is with QT.

I'm curious if you tried it out and what your thoughts are? @Anomalocaridid

Anomalocaridid commented:
@sgoudham I'm not sure I follow. In what way is this theme better without accent support? Do you mean in the sense that the accent support is redundant when using the system Qt theme instead?

I did try out using the system Qt theme with Prismlauncher. Originally, I used the Kvantum port and found I did not like how it looked as much as I did with this theme. However, I noticed the README.MD now recommends using qt5ct. So I tried it out just now and to my surprise, it looks exactly the same as this theme.

sgoudham commented:
> In what way is this theme better without accent support?

Felt like the on-hover highlights with the accents made it more difficult to read, but I'm wondering if I was encountering other issues now.

> However, I noticed the README.MD now recommends using qt5ct. So I tried it out just now and to my surprise, it looks exactly the same as this theme.

Yeah, I think Linux users will be better off using the QT theme 🤔
